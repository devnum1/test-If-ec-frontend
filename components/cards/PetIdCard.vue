// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="pet-id-card">
    <div class="id-card-container">
      <div class="peco-title">
        {{ $t('common.peco_title') }}
      </div>
      <div v-show="editCardPet" class="id-edit">
        <button class="id-edit__button" @click="handleClick">
          {{ $t('common.button.edit') }}
        </button>
      </div>
      <h4 class="pet-name font-weight-400" :class="editCardPet ? '' : 'mt-26'">
        <span>{{ get(petInfo, 'petName', '') }}</span>
        <span class="pet-gender">
          {{
            get(petInfo, 'gender') === constContract.pet_sex.FEMALE
              ? $t('common.chan')
              : $t('common.kun')
          }}
        </span>
      </h4>
      <div class="id-card__pet-info-container">
        <div class="pet-info-grid">
          <div class="info-row">
            <div class="card-icon">
              <i class="icon-fonts icon-fonts--cake" />
            </div>
            <div class="">
              {{ petBirthday }}
            </div>
          </div>
          <div class="info-row">
            <div class="card-icon">
              <i class="icon-fonts icon-fonts--dog-foot" />
            </div>
            <div class="">
              {{
                get(petInfo, 'petType') === constContract.pet_type.DOG
                  ? get(petInfo, 'petBreedDesc')
                  : $t('common.pet.cat')
              }}
            </div>
          </div>
          <div class="info-row">
            <div class="card-icon">
              <i class="icon-fonts icon-fonts--paper" />
            </div>
            <div class="">
              {{ compensRate }}
            </div>
          </div>
          <div class="info-row">
            <img
              class="card-logo"
              src="~/assets/images/logo/2line_bg_white.svg"
              height="36"
            />
          </div>
        </div>
        <div v-if="petImage" class="pet-image-container">
          <img class="pet-image" :src="petImage" />
        </div>
        <div v-else class="no-image-container">
          <img
            v-if="get(petInfo, 'petType') === constContract.pet_type.CAT"
            class="no-image cat-invert"
            src="~/assets/images/icon-cat.svg"
          />
          <img v-else class="no-image" src="~/assets/images/icon-dog.svg" />
        </div>
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
import contract from '~/constants/contract';
import { get } from 'lodash';
export default {
  props: {
    petInfo: {
      type: Object,
      default: () => ({}),
    },
    productInfo: {
      type: Object,
      default: () => ({}),
    },
    editCardPet: {
      type: Boolean,
      default: false,
    },
    policyNo: {
      type: String,
      default: '',
    },
    policyId: {
      type: String,
      default: '',
    },
    subscriberNo: {
      type: [String, Number],
      default: '',
    },
    noPetImage: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      petImage: null,
      constContract: contract,
    };
  },
  computed: {
    petBirthday() {
      const petBirthday = get(this.petInfo, 'birthDate', '______');

      return petBirthday.substring(0, 4) + '.' + petBirthday.substring(5, 7);
    },
    compensRate() {
      const plan = get(this.productInfo, 'planCode');
      if (!plan) {
        return 'Unknown';
      }

      return (
        (plan === contract.plan_code.SEVENTY ? 70 : 50) +
        this.$t('common.compens_rate')
      );
    },
  },
  async created() {
    // DP-0019画面ではペット画像を取得しない
    if (this.noPetImage) {
      return;
    }
    const petImageKey = get(this.petInfo, 'petPhotoDoc');

    if (petImageKey) {
      const res = await this.$store.dispatch('contracts/s3download', {
        imageKey: petImageKey,
      });

      if (res.result === 1) {
        this.petImage = res.docShareUrl;
      }
    }
  },

  methods: {
    get,
    handleClick() {
      this.$store.commit('contracts/setPolicyId', this.policyId);
      this.$emit('edit');
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/components/cards';
</style>
