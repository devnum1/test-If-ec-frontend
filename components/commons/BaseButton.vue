<template>
  <button
    type="submit"
    class="submit-button"
    :class="{
      [className]: !!className,
      disabled: disabled,
    }"
    :style="{
      width,
      height,
    }"
    :disabled="disabled"
    @click="$emit('click')"
  >
    <slot />
  </button>
</template>

<script>
export default {
  props: {
    className: {
      type: String,
      default: '',
    },
    width: {
      type: String,
      default: '',
    },
    height: {
      type: String,
      default: '',
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
.submit-button {
  color: #fff;
  display: flex;
  padding: 17px 20px;
  justify-content: center;
  align-items: center;
  border-radius: 47px;
  border: 1px solid #4b9590;
  background: #5fbbb5;
  box-shadow: 0 4px 0 0 #4b9590;
}
.disabled {
  background-color: $white;
  box-shadow: none;
  color: rgba($black, 0.15);
  border-color: rgba($black, 0.15);
}
</style>
