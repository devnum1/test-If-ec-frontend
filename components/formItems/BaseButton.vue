// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <button
    type="button"
    :class="[
      'btn btn--default',
      { 'btn--icon-right': iconRight },
      { 'btn--icon-plus': btnIconPlus },
      { 'btn--icon-pdf': btnIconPdf },
      { 'btn--icon-plus-label': btnIconPlusLabel },
      { 'btn--black': btnBlack },
      { 'btn--orange': btnOrange },
      { 'btn--green-outline': btnGreenOutline },
      { 'btn--black-outline': btnBlackOutline },
      btnClass,
      btnSize,
      btnFull,
    ]"
    :disabled="disabled"
    @click="callback($event)"
  >
    <i
      v-if="btnIconPlus || btnIconPlusLabel"
      class="icon-fonts icon-fonts--plus"
    />
    <i v-if="btnIconEmail" class="icon-fonts icon-fonts--mail mr-1 txt-green" />
    <span v-if="!btnIconPlus" class="btn__text" v-html="label" />
    <i v-if="iconRight" class="icon-fonts icon-fonts--arrow-right" />
    <img v-if="btnIconPdf" src="~/assets/images/pdf.svg" />
  </button>
</template>
<script>
export default {
  name: 'BaseButton',
  props: {
    label: {
      type: String,
      default: 'デフォルトボタン',
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    btnBlack: {
      type: Boolean,
      default: false,
    },
    btnOrange: {
      type: Boolean,
      default: false,
    },
    btnGreenOutline: {
      type: Boolean,
      default: false,
    },
    btnBlackOutline: {
      type: Boolean,
      default: false,
    },
    btnClass: {
      type: String,
      default: '',
    },
    iconRight: {
      type: Boolean,
      default: false,
    },
    btnIconPlus: {
      type: Boolean,
      default: false,
    },
    btnIconPlusLabel: {
      type: Boolean,
      default: false,
    },
    btnSize: {
      type: String,
      default: '',
    },
    btnFull: {
      type: String,
      default: '',
    },
    btnIconEmail: {
      type: Boolean,
      default: false,
    },
    btnIconPdf: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    callback: function (e) {
      this.$emit('click', e);
    },
  },
};
</script>
<style lang="scss" scoped>
@import '~/assets/scss/components/base-buttons';
</style>
