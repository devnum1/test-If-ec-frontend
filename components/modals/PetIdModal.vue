// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="pet-id-modal">
    <div class="modal__overlay">
      <div class="modal__container">
        <div class="modal__close-button">
          <div class="icon-delete" @click="closeModal()" />
        </div>
        <div class="modal__body modal-claim">
          <PetIdCard :pet-info="petInfo" />
          <div class="get-home">
            うちの子の写真を登録して、<br />
            うちの子だけのマイページにしよう！
          </div>
          <BaseButton
            btn-class="test"
            label="編集画面へ"
            btn-full="btn--full"
            icon-right
          />
        </div>
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
export default {
  data() {
    return {
      display: true,
      petInfo: {
        name: 'うめじまのうめ子',
        dob: '2018.01.01',
        type: 'スタッフォードシャーブルテリア',
        contract: '70％補償プラン',
        image: '',
        species: 0, // 1 => いぬ, 0 => ねこ
        gender: 0, // 1 => 男の子, 0 => 女の子
      },
    };
  },
  methods: {
    closeModal() {
      this.display = false;
      this.$emit('input', this.display);
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/components/modal';
.modal__container {
  padding: 32px 16px 26px 16px;
}
</style>
