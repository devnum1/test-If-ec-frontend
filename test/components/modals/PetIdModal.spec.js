import { shallowMount, config } from '@vue/test-utils';
import PetIdModal from '@/components/modals/PetIdModal.vue';
import PetIdCard from '@/components/cards/PetIdCard.vue';
import BaseButton from '@/components/formItems/BaseButton.vue';

config.mocks.$t = (key) => key;

describe('PetIdModal', () => {
  const petInfo = {
    name: 'うめじまのうめ子',
    dob: '2018.01.01',
    type: 'スタッフォードシャーブルテリア',
    contraception: 'なし',
    contract: '70％補償プラン',
    image: '',
    species: 0, // 1 => いぬ, 0 => ねこ
    gender: 1, // 1 => 男の子, 0 => 女の子
  };

  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(PetIdModal, {
      propsData: {},
      mocks: {},
      stubs: {},
      methods: {},
      computed,
      stubs: {
        // used to register custom components
        PetIdCard: PetIdCard,
        BaseButton: BaseButton,
      },
    });
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('to display PetIdCard exact', () => {
    const component = wrapper.findComponent(PetIdCard);
    expect(component.exists()).toBe(true);
  });

  // test xample: check trigger click item to run method
  test('Click close icon to call PetIdModal() method', () => {
    const closeBtn = wrapper.find('.icon-delete');
    wrapper.vm.closeModal = jest.fn();

    closeBtn.trigger('click');
    expect(wrapper.vm.closeModal).toHaveBeenCalled();
  });

  // test xample: check trigger click item to emit data value (this case: this.$emit('input', data))
  test('check closeModal() emits and sets value', () => {
    const closeBtn = wrapper.find('.icon-delete');

    closeBtn.trigger('click');

    expect(wrapper.emitted().input.length).toBe(1);
  });
});
