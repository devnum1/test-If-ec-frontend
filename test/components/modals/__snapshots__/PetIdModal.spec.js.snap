// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetIdModal match Snapshot 1`] = `
"<div class=\\"pet-id-modal\\">
  <div class=\\"modal__overlay\\">
    <div class=\\"modal__container\\">
      <div class=\\"modal__close-button\\">
        <div class=\\"icon-delete\\"></div>
      </div>
      <div class=\\"modal__body modal-claim\\">
        <div class=\\"pet-id-card\\">
          <div class=\\"id-card-container\\">
            <div class=\\"peco-title\\">
              common.peco_title
            </div>
            <div class=\\"id-edit\\" style=\\"display: none;\\"><button class=\\"id-edit__button\\">
                common.button.edit
              </button></div>
            <h4 class=\\"pet-name font-weight-400 mt-26\\"><span></span> <span class=\\"pet-gender\\">common.kun</span></h4>
            <div class=\\"id-card__pet-info-container\\">
              <div class=\\"pet-info-grid\\">
                <div class=\\"info-row\\">
                  <div class=\\"card-icon\\"><i class=\\"icon-fonts icon-fonts--cake\\"></i></div>
                  <div>
                    ____.__
                  </div>
                </div>
                <div class=\\"info-row\\">
                  <div class=\\"card-icon\\"><i class=\\"icon-fonts icon-fonts--dog-foot\\"></i></div>
                  <div>
                    common.pet.cat
                  </div>
                </div>
                <div class=\\"info-row\\">
                  <div class=\\"card-icon\\"><i class=\\"icon-fonts icon-fonts--paper\\"></i></div>
                  <div>
                    Unknown
                  </div>
                </div>
                <div class=\\"info-row\\"><img src=\\"~/assets/images/logo/2line_bg_white.svg\\" height=\\"36\\" class=\\"card-logo\\"></div>
              </div>
              <div class=\\"no-image-container\\"><img src=\\"~/assets/images/icon-dog.svg\\" class=\\"no-image\\"></div>
            </div>
          </div>
        </div>
        <div class=\\"get-home\\">
          うちの子の写真を登録して、<br> うちの子だけのマイページにしよう！
        </div> <button type=\\"button\\" class=\\"btn btn--default btn--icon-right test btn--full\\">
          <!---->
          <!----> <span class=\\"btn__text\\">編集画面へ</span> <i class=\\"icon-fonts icon-fonts--arrow-right\\"></i>
          <!----></button>
      </div>
    </div>
  </div>
</div>"
`;
